examples/platform/linux/CommissionerMain.cpp

```diff
@@ -110,14 +113,50 @@ class MyCommissionerCallback : public CommissionerCallback
     }
 };
 
+AutoCommissioner gAutoCommissioner;
+
+class MyCredsIssuer : public ExampleOperationalCredentialsIssuer
+{
+    CHIP_ERROR GenerateNOCChain(const ByteSpan & csrElements, const ByteSpan & csrNonce, const ByteSpan & attestationSignature,
+                                const ByteSpan & attestationChallenge, const ByteSpan & DAC, const ByteSpan & PAI,
+                                Callback::Callback<OnNOCChainGeneration> * onCompletion) override
+    {
+        ChipLogError(Controller, "---------------------------MyCredsIssuer ");
+        // add parsing here
+        const ByteSpan & attestationElements = gAutoCommissioner.GetCommissioningParameters().GetAttestationElements().Value();
+        ChipLogError(Controller, "MyCredsIssuer attestationElements size %d", static_cast<int>(attestationElements.size()));
+
+        ByteSpan certificationDeclarationSpan;
+        ByteSpan attestationNonceSpan;
+        uint32_t timestampDeconstructed;
+        ByteSpan firmwareInfoSpan;
+        DeviceAttestationVendorReservedDeconstructor vendorReserved;
+
+        CHIP_ERROR err = DeconstructAttestationElements(attestationElements, certificationDeclarationSpan, attestationNonceSpan,
+                                                        timestampDeconstructed, firmwareInfoSpan, vendorReserved);
+        if (err != CHIP_NO_ERROR)
+        {
+            ChipLogError(Controller, "Failed to parse attestation elements");
+        }
+        else
+        {
+            ChipLogError(Controller, "Parse attestation elements success");
+        }
+
+        return ExampleOperationalCredentialsIssuer::GenerateNOCChain(csrElements, csrNonce, attestationSignature,
+                                                                     attestationChallenge, DAC, PAI, onCompletion);
+    }
+};
+
 DeviceCommissioner gCommissioner;
 CommissionerDiscoveryController gCommissionerDiscoveryController;
 MyCommissionerCallback gCommissionerCallback;
 MyServerStorageDelegate gServerStorage;
-ExampleOperationalCredentialsIssuer gOpCredsIssuer;
+MyCredsIssuer gOpCredsIssuer;
 NodeId gLocalId = kMaxOperationalNodeId;
 Credentials::GroupDataProviderImpl gGroupDataProvider;
-AutoCommissioner gAutoCommissioner;
+Credentials::PartialDACVerifier gPartialDACVerifier;
+FabricId gFabricId = 1;
```

Comment:
This likely does not belong globally. Suggest using a constant

Reply:
I think a lot of the objects in CommissionerMain used for commissioning will need to be overridden by specific implementations. I suggest moving this to a separate "cleanup" issue.

Reply:
Are you saying this is OK because it's in "CommissionerMain.cpp" which only applies to some odd examples? All other chip examples allow setting fabric ID from command line.

Reply:
added command line option for fabric Id